app/validation.go

```diff
@@ -79,6 +91,38 @@ func checkForUndefinedBOSHProperties(label string, properties map[string]string,
 	return allErrs
 }
 
+// checkParentsOfUndefined walks the chain of parents for `p` from the
+// bottom up and checks if any of them exist. The elements of the
+// chain are separated by dots.
+func checkParentsOfUndefined(p string, bosh propertyDefaults) bool {
+	at := strings.LastIndex(p, ".")
+
+	for at >= 0 {
+		// While there is a dot in the property name we have a
+		// parent to check the existence of
+
+		tail := p[at:]
+		parent := strings.TrimSuffix(p, tail)
```

Comment:
Ah. I thought the other way read better, but they're equivalent, so if you think that's clearer it's fine.